docs/docs/explanations/starlark-idempotent-runs.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Starlark idempotent runs
+sidebar_label: Starlark idempotent runs
+---
+
+Background
+----------
+
+Did you notice those lines in the `kurtosis run` command output?
+```console
+SKIPPED - This instruction has already been run in this enclave
+```
+To reduce execution time, Kurtosis tries to optimize each run of a Starlark package based on what has already been run 
```

Comment:
```suggestion
The reason this happens is because Kurtosis will optimize each run of a Starlark package based on what has already been run in a given enclave, thus reducing execution time and resources.
```